src/components/check-your-email/check-your-email-service.ts

```diff
@@ -2,34 +2,29 @@ import { getRequestConfig, Http, http } from "../../utils/http";
 import { API_ENDPOINTS, HTTP_STATUS_CODES } from "../../app.constants";
 import { CheckYourEmailServiceInterface } from "./types";
 import { AxiosResponse } from "axios";
+import {UpdateInformationInput, UpdateInformationSessionValues} from "../../utils/types";
 
 export function checkYourEmailService(
   axios: Http = http
 ): CheckYourEmailServiceInterface {
   const updateEmail = async function (
-    accessToken: string,
-    existingEmailAddress: string,
-    replacementEmailAddress: string,
-    code: string,
-    sourceIp: string,
-    sessionId: string,
-    persistentSessionId: string,
-    userLanguage: string
+      updateInput : UpdateInformationInput,
+    sessionDetails: UpdateInformationSessionValues,
   ): Promise<boolean> {
     const { status }: AxiosResponse = await axios.client.post(
       API_ENDPOINTS.UPDATE_EMAIL,
       {
-        existingEmailAddress,
-        replacementEmailAddress,
-        otp: code,
+        existingEmailAddress : updateInput.email,
+        replacementEmailAddress : updateInput.updatedValue,
+        otp: updateInput.otp,
       },
       getRequestConfig(
-        accessToken,
+        sessionDetails.accessToken,
         [HTTP_STATUS_CODES.NO_CONTENT, HTTP_STATUS_CODES.BAD_REQUEST],
-        sourceIp,
-        persistentSessionId,
-        sessionId,
-        userLanguage
+          sessionDetails.sourceIp,
+          sessionDetails.persistentSessionId,
+          sessionDetails.sessionId,
+          sessionDetails.userLanguage
       )
     );
```

Comment:
I think using a `sessionDetails` object is a good improvement - maybe we should extend this approach to all the change details components which are calling `getRequestConfig()` for consistency (probably in a separate ticket). If we do this we could also update the `getRequestConfig()` function to accept the parameters differently, for example, possibly using the spread operator: `getRequestConfig(validationStatues, ...sessionDetails)`

Reply:
yes :) will look into raising a separate ticket